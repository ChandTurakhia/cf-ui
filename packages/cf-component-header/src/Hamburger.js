import React from 'react';
import PropTypes from 'prop-types';
import Icon from 'cf-component-icon';
import { createComponent } from 'cf-style-container';

const Wrapper = createComponent(
  ({ theme }) => ({
    fontSize: '1.3rem',
    borderRight: `1px solid ${theme.color.smoke}`,
    padding: '0.9em 1em 1em 1em',
    userSelect: 'none',
    cursor: 'pointer',
    tablet: {
      display: 'none'
    }
  }),
  'div',
  ['onClick']
);

const Hamburger = ({ onClick, label }) => (
  <Wrapper onClick={onClick}><Icon label={label} type="hamburger" /></Wrapper>
);

Hamburger.propTypes = {
  label: PropTypes.string,
  onClick: PropTypes.func.isRequired
};

Hamburger.defaultProps = {
  label: 'Open the menu'
};

export default Hamburger;
