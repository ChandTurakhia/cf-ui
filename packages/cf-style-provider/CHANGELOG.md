# Change Log

All notable changes to this project will be documented in this file.
See [standard-version](https://github.com/conventional-changelog/standard-version) for commit guidelines.

<a name="2.2.3"></a>
## [2.2.3](https://github.com/koddsson/cf-ui/compare/cf-style-provider@2.2.2...cf-style-provider@2.2.3) (2017-07-31)




<a name="2.2.2"></a>
## [2.2.2](https://github.com/cloudflare/cf-ui/compare/cf-style-provider@2.2.1...cf-style-provider@2.2.2) (2017-07-19)




<a name="2.2.0"></a>
# [2.2.0](https://github.com/sejoker/cf-ui/compare/cf-style-provider@2.1.2...cf-style-provider@2.2.0) (2017-07-12)


### Features

* **cf-style-provider:** allow passing selectPrefix in cf-style-provider ([3bc1ba9](https://github.com/sejoker/cf-ui/commit/3bc1ba9))




<a name="2.1.4"></a>
## [2.1.4](https://github.com/koddsson/cf-ui/compare/cf-style-provider@2.1.3...cf-style-provider@2.1.4) (2017-07-10)




<a name="2.1.3"></a>
## [2.1.3](https://github.com/koddsson/cf-ui/compare/cf-style-provider@2.1.2...cf-style-provider@2.1.3) (2017-07-10)




<a name="2.1.2"></a>
## [2.1.2](https://github.com/cloudflare/cf-ui/compare/cf-style-provider@2.1.0...cf-style-provider@2.1.2) (2017-07-06)




<a name="2.1.1"></a>
## [2.1.1](https://github.com/koddsson/cf-ui/compare/cf-style-provider@2.1.0...cf-style-provider@2.1.1) (2017-07-06)




<a name="2.1.0"></a>
# [2.1.0](https://github.com/cloudflare/cf-ui/compare/cf-style-provider@2.0.2...cf-style-provider@2.1.0) (2017-07-05)


### Features

* **cf-style-provider:** Print warnings in the dev mode if custom media queries are used ([b8a67c6](https://github.com/cloudflare/cf-ui/commit/b8a67c6))
* **cf-style-provider:** Remove monolithic mode from the dev configuration ([427bf80](https://github.com/cloudflare/cf-ui/commit/427bf80))
* **cf-style-provider:** Sort media queries, add better README ([d89f7fe](https://github.com/cloudflare/cf-ui/commit/d89f7fe))
* **cf-style-provider:** Update snapshots - they use atomic classNames now ([1a42284](https://github.com/cloudflare/cf-ui/commit/1a42284))
