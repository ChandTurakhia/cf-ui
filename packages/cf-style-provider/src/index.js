import createRenderer from './createRenderer';
import felaSnapshot from './felaSnapshot';
import felaTestContext from './felaTestContext';
import StyleProvider from './StyleProvider';

export { createRenderer, felaSnapshot, felaTestContext, StyleProvider };
