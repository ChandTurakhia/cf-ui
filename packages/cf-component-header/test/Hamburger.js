import React from 'react';
import renderer from 'react-test-renderer';
import Hamburger from '../src/Hamburger';
import { felaSnapshot } from 'cf-style-provider';

test('should render', () => {
  const snapshot = felaSnapshot(<Hamburger />);
  expect(snapshot.component).toMatchSnapshot();
  expect(snapshot.styles).toMatchSnapshot();
});
