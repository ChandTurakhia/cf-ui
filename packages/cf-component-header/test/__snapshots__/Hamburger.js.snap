// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div
  className="a b c d e f"
  onClick={[Function]}
>
  <i
    className="cf-icon cf-icon--hamburger"
    role={undefined}
  >
    <span
      className="cf-icon__label"
    >
      Open the menu
    </span>
  </i>
</div>
`;

exports[`should render 2`] = `
"
.a {
  font-size: 1.3rem
}

.b {
  border-right: 1px solid #e0e0e0
}

.c {
  padding: 0.9em 1em 1em 1em
}

.d {
  user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none
}

.e {
  cursor: pointer
}

@media (min-width: 47.2em) {
  .f {
    display: none
  }
}
"
`;
